src/main/java/Util/Ui.java

```diff
@@ -0,0 +1,244 @@
+                        throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+                    }
+                    showDeadline(input);
+                    continue;
+                }
+                if(input.substring(0,5).equals("event")) {
+                    try {
+                        input.substring(6,7);
+                    } catch (Exception e){
+                        throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+                    }
+                    showEvent(input);
+                    continue;
+                }
+            }
+        }
+    }
+
+    /**
+     * shows Welcome message when called
+     */
+    public void showWelcome() {
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+    }
+
+    /**
+     * shows Exit message when called
+     */
+    public void showExit() {
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+    /**
+     * shows done message when called and performs rewrite on storage
+     * and done on that task in TaskList
+     */
+    public void showDone(String input) throws Exception{
+        int itemIndex = Integer.parseInt(input.substring(5)) - 1;
+        // close if item number don't exist
+        if((itemIndex + 1) > tasksList.getTaskList().size()) {
+            System.out.println("failed to done item, please try again");
+        } else {
+            tasksList.markTaskDone(itemIndex);
+            storage.rewriteWriter(tasksList);
+        }
+    }
+
+    /**
+     * shows delete message when called and performs rewrite on storage
+     * and delete that task from TaskList
+     */
+    public void showDelete(String input) throws Exception{
+        int itemIndex = Integer.parseInt(input.substring(7)) - 1;
+        if((itemIndex + 1) > tasksList.getTaskList().size()) {
+            System.out.println("failed to delete item, please try again.");
+        } else {
+            System.out.println("Noted. I've removed this task:");
+            System.out.println("  " + tasksList.getTaskList().get(itemIndex));
+            tasksList.getTaskList().remove(itemIndex);
+            storage.rewriteWriter(tasksList);
+            System.out.println("Now you have " + tasksList.getTaskList().size() + " in the list.");
+        }
+    }
+
+    /**
+     * shows todo message when called and performs read on storage
+     * and appends a todo task in TaskList
+     */
+    public void showTodo(String input){
+        String inputAdd = input.substring(5);
+        Todo todo = new Todo(inputAdd, "");
+        tasksList.addTask(todo);
+        storage.writeTodo(todo);
+        System.out.println("  " + todo);
+    }
+
+    /**
+     * shows deadline message when called and performs read on storage
+     * and appends a deadline task in TaskList
+     */
+    public void showDeadline(String input) throws Exception{
```

Comment:
showing and adding should be two separate things. Also you can just add it in first, then call a toString on the task, and then all these functions can be collapsed into one.